python/python/bystro/proteomics/fragpipe_tandem_mass_tag.py

```diff
@@ -0,0 +1,53 @@
+"""Load and prep fragpipe tandem mass Tag datasets."""
+
+from dataclasses import dataclass
+
+import pandas as pd
+
+ABUNDANCE_COLS = ["Index", "NumberPSM", "ProteinID", "MaxPepProb", "ReferenceIntensity"]
+ANNOTATION_COLS = ["plex", "channel", "sample", "sample_name", "condition", "replicate"]
+
+
+@dataclass(frozen=True)
+class TandemMassTagDataset:
+    """Represent a Fragpipe Tandem Mass Tag dataset."""
+
+    abundance_df: pd.DataFrame
+    annotation_df: pd.DataFrame
+
+
+def _check_df_cols(df: pd.DataFrame, expected_cols: list[str]) -> None:
+    actual_cols = df.columns
+    if not all(x == y for x, y in zip(expected_cols, actual_cols, strict=False)):
+        err_msg = (
+            f"expected dataframe to begin with cols: {expected_cols}, got cols: {actual_cols} instead."
+        )
+        raise ValueError(err_msg)
+
+
+def _prep_abundance_df(abundance_df: pd.DataFrame) -> pd.DataFrame:
+    """Prep abundance_df, setting index and normalizing abundances by ReferenceIntensity."""
+    _check_df_cols(abundance_df, ABUNDANCE_COLS)
+    abundance_df = abundance_df.set_index("Index")
+    first_sample_column_idx = abundance_df.columns.to_list().index("ReferenceIntensity") + 1
+    sample_columns = abundance_df.columns[first_sample_column_idx:]
+    for sample_column in sample_columns:
+        abundance_df[sample_column] -= abundance_df.ReferenceIntensity
```

Comment:
Is this a standard normalization step? I thought they were normalized in a different way, but just checking. As a side note, is it possible we want to support different possible normalization schemes as a small analysis step and leave them unnormalized for the basic report people will search through?

Reply:
If we don't normalize in this way, then ReferenceIntensity explains something like 90% of the variance of each column.  I'm of course open to doing it in some other way, but what would we do instead?

Reply:
Looks like there's quite a few normalization methods - just wanted to be sure we were going with a standard choice, would be worth conferring with someone else about this. Might also make more sense as a downstream analysis step that happens after annotation? Are we reporting variance in the annotation?